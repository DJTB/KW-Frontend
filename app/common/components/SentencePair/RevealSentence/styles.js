import styled from "styled-components";

import P from "common/components/P";
import Icon from "common/components/Icon";

import { gutter } from "common/styles/layout";
import { borderRadius } from "common/styles/sizing";
import { greyDark, transparent } from "common/styles/colors";
import { fastEaseQuad, midEaseQuad } from "common/styles/animation";

export const Wrapper = styled.div`
  ${gutter()}
`;

export const RevealIcon = styled(Icon)`
  position: absolute;
  top: 50%;
  left: 50%;
  opacity: 1;
  transition: opacity ${fastEaseQuad};
  transform: translate(-50%, -50%);
  z-index: 1;
  color: ${greyDark};
  pointer-events: none;
`;

export const Sentence = P.extend`
  position: relative;
  display: inline-flex;
  font-size: 1.1em;
  font-style: italic;
  line-height: 1.2;
  transition: all ${fastEaseQuad};
  border-radius: ${borderRadius};
  /* blur effect */
  color: ${transparent};
  text-shadow: 0 0 0.8em rgba(0, 0, 0, 0.4);
  &:hover,
  &:active,
  &:focus {
    transition: all ${midEaseQuad};
    outline: none;
    color: ${greyDark};
    text-shadow: none;
    & ${RevealIcon} {
      transition: opacity ${midEaseQuad};
      opacity: 0;
    }
  }
`;
