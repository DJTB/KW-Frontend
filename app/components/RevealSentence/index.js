import React from 'react';
import PropTypes from 'prop-types';

import { Wrapper, Sentence, RevealIcon } from './styles';

RevealSentence.propTypes = {
  sentence: PropTypes.string.isRequired,
};

function RevealSentence({ sentence }) {
  return (
    <Wrapper tabIndex="0">
      <Sentence>{sentence}</Sentence>
      <RevealIcon name="EYE" title="View masked sentence" />
    </Wrapper>
  );
}

export default RevealSentence;
