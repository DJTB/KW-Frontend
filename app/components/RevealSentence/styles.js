import styled from 'styled-components';
import { rgba } from 'polished';

import P from 'base/P';
import Icon from 'components/Icon';

import { gutter } from 'shared/styles/layout';
import { borderRadius } from 'shared/styles/sizing';
import { whiteLight, greyDark, transparent } from 'shared/styles/colors';
import { fastEaseQuad, midEaseQuad } from 'shared/styles/animation';

export const Sentence = P.extend`
  font-size: 1.1em;
  font-style: italic;
  line-height: 1.2;
  color: ${greyDark};
  transition: all ${fastEaseQuad};
  border-radius: ${borderRadius};
  background-color: ${rgba(greyDark, 0.95)};
  @supports( filter: blur() ) {
    background-color: ${transparent};
    filter: blur(5px);
  }
`;

export const RevealIcon = styled(Icon)`
  position: absolute;
  ${gutter({ prop: 'margin', position: 'top', mod: 0.5 })}
  left: 50%;
  top: 0;
  opacity: 1;
  transition: opacity ${fastEaseQuad};
  transform: translate(-50%, 0);
  z-index: 1;
  color: ${whiteLight};
  @supports( filter: blur() ) {
    color: ${greyDark};
  }
`;

export const Wrapper = styled.div`
  position: relative;
  align-self: flex-start;

  &:hover,
  &:active,
  &:focus {
    & ${Sentence} {
      transition: all ${midEaseQuad};
      outline: none;
      color: greyDark;
      background-color: transparent;
      @supports( filter: blur() ) {
        filter: blur(0px);
      }
    }
    & ${RevealIcon} {
      transition: opacity ${midEaseQuad};
      opacity: 0;
    }
  }
`;
