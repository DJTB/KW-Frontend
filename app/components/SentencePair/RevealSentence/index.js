import React from "react";
import PropTypes from "prop-types";

import { Sentence, RevealIcon } from "./styles";

RevealSentence.propTypes = {
  sentence: PropTypes.string.isRequired,
};

function RevealSentence({ sentence }) {
  return (
    <div>
      <Sentence tabIndex="0">
        {sentence}
        <RevealIcon name="EYE" />
      </Sentence>
    </div>
  );
}

export default RevealSentence;
