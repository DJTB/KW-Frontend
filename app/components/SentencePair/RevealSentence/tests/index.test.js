import 'jest-styled-components';
import React from 'react';
import { render } from 'enzyme';
import RevealSentence from '../index';

describe('<RevealSentence />', () => {
  it('should match baseline snapshot', () => {
    expect(render(<RevealSentence />)).toMatchSnapshot();
  });
});
