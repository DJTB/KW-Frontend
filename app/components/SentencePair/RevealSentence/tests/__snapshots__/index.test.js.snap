// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RevealSentence /> should match baseline snapshot 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  position: relative;
  width: 1.5em;
  height: 1.5em;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c6 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  fill: currentColor;
}

.c2 {
  padding: 0.225rem;
  font-size: 15.3px;
  padding: 0.225rem;
  margin-top: 0;
  margin-bottom: 0;
  line-height: 1.3;
  margin-left: 0;
  margin-right: 0;
  font-size: 1.1em;
  font-style: italic;
  line-height: 1.2;
  color: #616161;
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  border-radius: .2em;
  background-color: rgba(97,97,97,0.95);
}

.c4 {
  position: absolute;
  margin-top: 0.1125rem;
  left: 50%;
  top: 0;
  opacity: 1;
  -webkit-transition: opacity 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: opacity 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  z-index: 1;
  color: #fafafa;
}

.c0 {
  position: relative;
}

.c0:hover .c1,
.c0:active .c1,
.c0:focus .c1 {
  -webkit-transition: all 375ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: all 375ms cubic-bezier(0.455,0.030,0.515,0.955);
  outline: none;
  color: #616161;
  background-color: transparent;
}

.c0:hover .c3,
.c0:active .c3,
.c0:focus .c3 {
  -webkit-transition: opacity 375ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: opacity 375ms cubic-bezier(0.455,0.030,0.515,0.955);
  opacity: 0;
}

@media (min-width:600px) {
  .c2 {
    padding: 0.33325rem;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c2 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c2 {
    font-size: 17.4px;
  }
}

@media (min-width:600px) {
  .c2 {
    padding: 0.33325rem;
  }
}

@supports (-webkit-filter:blur() ) or (filter:blur() ) {
  .c2 {
    background-color: rgba(0,0,0,0);
    -webkit-filter: blur(5px);
    filter: blur(5px);
  }
}

@media (min-width:600px) {
  .c4 {
    margin-top: 0.166625rem;
  }
}

@supports (-webkit-filter:blur() ) or (filter:blur() ) {
  .c4 {
    color: #616161;
  }
}

@supports (-webkit-filter:blur() ) or (filter:blur() ) {
  .c0:hover .c1,
  .c0:active .c1,
  .c0:focus .c1 {
    -webkit-filter: blur(0px);
    filter: blur(0px);
  }
}

<div
  class="c0"
  tabindex="0"
>
  <p
    class="c1 c2"
  />
  <span
    class="c3 c4 c5"
    color="currentColor"
  >
    <svg
      class="c6"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      title="EYE"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
      />
    </svg>
  </span>
</div>
`;
